flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/datastream/DataStream.java

```diff
@@ -256,9 +256,11 @@ public ExecutionConfig getExecutionConfig() {
 		DataStream<OUT> returnStream = this.copy();
 
 		for (DataStream<OUT> stream : streams) {
-			for (DataStream<OUT> ds : stream.unionedStreams) {
-				validateUnion(ds.getId());
-				returnStream.unionedStreams.add(ds.copy());
+			if (stream != null) {
```

Comment:
Hi,
you may look at this problem with my test above.
As you see, if I ignore the null, code "temp2.union(temp1)" will just copy a new DataStream.
And this new DataStream has absolutely no change comparing with temp2.
So, If the temp2 is good, the new DataStream is good too.
And the test or other case similar to this test will execute successfully.
Instead, there just will be a error.